packages/app/components/token-picker/TokenPicker.tsx

```diff
@@ -93,6 +117,70 @@ export const TokenPicker = ({
     if (chainId) setCommonTokens(TOKEN_PICKER_COMMON_TOKENS[chainId]);
   }, [chainId]);
 
+  useEffect(() => {
+    if (isConnected && tokenListWithBalances) {
+      setFilteredTokenList(tokenListWithBalances);
+    } else {
+      setFilteredTokenList(tokenList);
+    }
+  }, [isConnected, tokenList, tokenListWithBalances]);
+
+  const onClearSearch = () => {
+    tokenListSearchCleanup();
+    tokenListCleanup();
+  };
+
+  useEffect(() => {
+    if (debouncedQuery) {
+      const newList =
+        isConnected && tokenListWithBalances
+          ? tokenListWithBalances
+          : tokenList;
+
+      const filteredList = newList.filter(
+        (token) =>
+          token.symbol.toLowerCase().includes(debouncedQuery.toLowerCase()) ||
+          token.name.toLowerCase().includes(debouncedQuery.toLowerCase())
+      );
```

Comment:
makes sense to filter per address, sometimes people paste the address, it's handy so you are sure it is the token you're looking for.

Reply:
You are right, this was a bug. Fixed